ranger-companion.lic

```diff
@@ -0,0 +1,80 @@
+=begin
+  Documentation: https://elanthipedia.play.net/Lich_script_repository#ranger-companion
+=end
+
+custom_require.call(%w[common])
+
+no_pause_all
+
+class Companion
+  include DRC
+
+  def initialize
+    # TODO: Add raccoon messaging
+    case bput('whistle for companion', 'Your lips are too dry', 'wolf scrambles in', 'You whistle a merry tune', 'wolf perks', 'You whistle loudly for your companion', 'purse your lips to call')
+    when 'purse your lips to call'
+      exit
+    end
+
+    while line = get
+      waitrt?
+      # Dismiss wolf if it grows past the baby stage
+      if line =~ /^.*(young|full-grown) wolf (begins|howls|paws|looks|perks|sits|pants|stands|carefully)/
+        bput('signal companion to hunt', 'suddenly catches a scent', 'is too young to hunt', 'Your companion is not', 'You have no companion', 'snapping your fingers')
+        exit
+      end
+      # A baby wolf stands up then paces back and forth nervously.
+      # A baby wolf paces back and forth.
+      if line =~ /^.*(wolf|then) paces back and forth*/
+        case bput('pet wolf', 'You pet your baby wolf', 'Touch what', 'wolf shies away from you')
+        # Matches when there are 2 wolves in the room
+        when 'wolf shies away from you'
+          bput('pet second wolf', 'You pet your baby wolf', 'Touch what', 'wolf shies away from you')
+        end
+      end
+      if line =~ /^.*raccoon paces back and forth*/
+        bput('pet raccoon', 'You pet', 'Touch what')
+      end
+      if line =~ /^A .* wolf begins to whimper./
+        Script.running.find_all { |s| !s.paused? && !s.no_pause_all }.each(&:pause)
+        waitrt?
+        bput('stow left', 'Stow what', 'You put')
+        case bput('get my milk', 'You get', 'You are already holding', 'What were you referring to')
+        when 'What were you referring to'
+          bput('signal compaion to sleep', 'wolf wanders off to find', 'Your companion is not', 'You have no companion', 'snapping your fingers')
+          exit
+        end
+        case bput('feed my milk to wolf', 'The baby wolf greedily drinks', "It doesn't seem hungry", "You can't feed with", 'wolf shies away from you')
+        when 'wolf shies away from you'
+          bput('feed my milk to second wolf', 'The baby wolf greedily drinks', "It doesn't seem hungry", "You can't feed with", 'wolf shies away from you')
```

Comment:
Should this `feed` also check for `"You can't feed with"` and exit in that case?

If so, I recommend making a `feed` helper function and having it called with a different arg in this case